### PR TITLE
Fix `rewrite_base_url`: don't discard query params

### PR DESCRIPTION
**Pull request by jjant**
## Description
The previous version of this function discarded query params, which meant that if our model had a `@httpQuery` bound input, those would not be sent to the server and the requests would fail.

## Testing
Ran existing tests

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: examples/pokemon-service-common/tests/plugins_execution_order.rs [hunk 1]
```diff
@@ -64,10 +64,7 @@ struct SentinelPlugin {
 
 impl SentinelPlugin {
     pub fn new(name: &'static str, output: Arc<Mutex<Vec<&'static str>>>) -> Self {
-        Self {
-            name,
-            output: output,
-        }
+        Self { name, output }
```
Comment (jjant) [inline examples/pokemon-service-common/tests/plugins_execution_order.rs:67]: Unrelated clippy lint fix.

Comment (david-perez) [reply examples/pokemon-service-common/tests/plugins_execution_order.rs:67]: Isn't clippy run in CI already? https://github.com/awslabs/smithy-rs/blob/c3691087d434a886a259399cea4a9d15b0780a06/examples/Makefile#L23-L24

File: examples/pokemon-service-common/src/lib.rs [hunk 2]
```diff
@@ -33,11 +37,16 @@ const PIKACHU_JAPANESE_FLAVOR_TEXT: &str =
...
+    authority: Authority,
+) -> impl Fn(Request) -> Request + Clone {
     move |mut req| {
         let http_req = req.http_mut();
-        let uri = format!("{base_url}{}", http_req.uri().path());
-        *http_req.uri_mut() = uri.parse().unwrap();
+        let mut uri_parts = http_req.uri().clone().into_parts();
+        uri_parts.authority = Some(authority.clone());
+        uri_parts.scheme = Some(scheme.clone());
+        *http_req.uri_mut() = Uri::from_parts(uri_parts).expect("valid uri");
```
Comment (david-perez) [inline examples/pokemon-service-common/src/lib.rs:49]: You're being misguided here and in the below usages by `.expect()`'s unfortunate naming into thinking that the message here is expressing what you expect, whereas this is supposed to be an error message :) See https://github.com/chris-morgan/rfcs/blob/unwrap_or_panic/text/0000-rename-expect.md#rationale-and-alternatives:

> [...] the name expect isn’t great on a couple of axes:
> It guides you in the direction of writing the message back to front, expressing what you expected to be the case, whereas it is supposed to be an error message when the expectation failed, meaning what went wrong, which is often expressed as the exact opposite.

**Review by david-perez — APPROVED** [review]

File: examples/pokemon-service-common/tests/plugins_execution_order.rs (hunk 1 above)
Comment (jjant) [reply examples/pokemon-service-common/tests/plugins_execution_order.rs:67]: Yeah, I guess I have a newer version locally.

**Pull request merged**